cmd/ignite/run/start.go

```diff
@@ -47,9 +51,41 @@ func Start(so *startOptions) error {
 		return err
 	}
 
+	if err := waitForSSH(so.vm, 10); err != nil {
+		return err
+	}
+
 	// If starting interactively, attach after starting
 	if so.Interactive {
 		return Attach(so.attachOptions)
 	}
 	return nil
 }
+
+func waitForSSH(vm *ignite.VM, seconds int) error {
+	// When --ssh is enabled, wait until SSH service started on port 22 at most N seconds
+	ssh := vm.Spec.SSH
+	if ssh != nil && ssh.Generate {
+		if len(vm.Status.IPAddresses) > 0 {
+			addr := vm.Status.IPAddresses[0].String() + ":22"
+			var err error
+			for i := 0; i < seconds*10; i++ {
+				conn, dialErr := net.DialTimeout("tcp", addr, 100*time.Millisecond)
```

Comment:
This could make this more clear:
```go
perSecond := 10
delay := time.Second / perSecond

for i := 0; i < seconds*perSecond; i++ {
    conn, dialErr := net.DialTimeout("tcp", addr, delay)
    ...
    time.Sleep(delay)
}
```